### PR TITLE
add electronics/coulombs law module

### PR DESCRIPTION
**Pull request by alirezasariri78**
# Pull Request Template

## Description

Please include a summary of the change and which issue (if any) is fixed.
A brief description of the algorithm and your implementation method can be helpful too. If the implemented method/algorithm is not so
well-known, it would be helpful to add a link to an article explaining it with more details.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I ran bellow commands using the latest version of **rust nightly**.
- [x] I ran `cargo clippy --all -- -D warnings` just before my last commit and fixed any issue that was found.
- [x] I ran `cargo fmt` just before my last commit.
- [x] I ran `cargo test` just before my last commit and all tests passed.
- [x] I added my algorithm to the corresponding `mod.rs` file within its own folder, and in any parent folder(s).
- [x] I added my algorithm to `DIRECTORY.md` with the correct link.
- [x] I checked `COUNTRIBUTING.md` and my code follows its guidelines.

Please make sure that if there is a test that takes too long to run ( > 300ms), you `#[ignore]` that or
try to optimize your code or make the test easier to run. We have this rule because we have hundreds of
tests to run; If each one of them took 300ms, we would have to wait for a long time.

### CODE REVIEW

File: src/matrix/search_in_a_sorted_matrix.rs
Comment (vil02) [inline src/matrix/search_in_a_sorted_matrix.rs:1]: Is this file relevant in this PR? If not, please remove it.

File: src/electronics/coulombs_law.rs [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        return Ok(format!(
+            "charge2: {}",
+            calculate_charge(charge1, force, distance)
+        ));
+    }
+    Ok(format!(
+        "distance: {}",
+        calculate_distance(charge_product, force)
+    ))
+}
```
Comment (vil02) [inline src/electronics/coulombs_law.rs:58]: Is this function really needed? If you insist on having it, please consider using `Optional` as inputs.

**Review by vil02 — CHANGES_REQUESTED** [review]

File: src/matrix/search_in_a_sorted_matrix.rs
Comment (alirezasariri78) [reply src/matrix/search_in_a_sorted_matrix.rs:1]: this module is located in python repository : https://github.com/TheAlgorithms/Python/blob/master/matrix/searching_in_sorted_matrix.py

File: src/electronics/coulombs_law.rs (hunk 1 above)
Comment (alirezasariri78) [reply src/electronics/coulombs_law.rs:58]: based on electronic modules in python repository : https://github.com/TheAlgorithms/Python/blob/master/electronics/coulombs_law.py

Comment (alirezasariri78) [reply src/electronics/coulombs_law.rs:58]: using optional would not cause boiler plate ?
in this situation i should double check value of inputs (none or some and sign of inputs)

**Pull request closed without merging**